### ISSUE TITLE
GPUImageStillCamera Slow Photo Taking

### ISSUE THREAD

**Issue by ychw** [BradLarson/GPUImage#1841]
I found it's very slow to take a picture with GPUImageStillCamera. On an iPhone 6, it usually takes 0.5 - 1 second for <code>capturePhotoAsImageProcessedUpToFilter:withCompletionHandler:</code> to finish. Following is my code:

``` obj-c
    self.cropFilter = [[GPUImageCropFilter alloc] initWithCropRegion:CGRectMake(0.0f, 0.0f, 1.0f, 0.75f)];

    self.resizeFilter = [[GPUImageFilter alloc] init];
    [self.resizeFilter forceProcessingAtSizeRespectingAspectRatio:CGSizeMake(2000, 2000)];

    [self.gpuImageStillCamera addTarget:self.cropFilter];
    [self.cropFilter addTarget:self.resizeFilter];
    [self.resizeFilter addTarget:self.gpuImageView];

    // following call takes 0.5 to 1 sec to come back on an iPhone 6
    [self.gpuImageStillCamera capturePhotoAsImageProcessedUpToFilter:self.resizeFilter withCompletionHandler:^(UIImage *processedImage, NSError *error) {
    // ....
    }
```

After further investigation, I found method <code>captureStillImageAsynchronouslyFromConnection: completionHandler:</code> from <code>AVCaptureStillImageOutput</code> takes most of the time. 

Is there a way to make photo taking on GPUImage more instant?

Thanks

**Comment by holgersindbaek** [comment]
+1 for this. If it takes so long time to take an image (iPhone 5c here takes 2 seconds), then I can't use GPUimage.

**Comment by holgersindbaek** [comment]
@ychw Did you find a solution for this?

**Comment by ychw** [comment]
@holgersindbaek: no, I didn't. But I benchmarked the default image picker controller and found it also takes some time to take a picture. GPUImage is still a little slower, but not a lot. So I just disabled user input within that 0.5 to 1 second while a photo is being taken. And it works out just fine.
